### PR TITLE
Create aggregators article

### PR DESCRIPTION
**Pull request by robertagora**
An article about aggregating small deals.  This should appear after direct dealmaking on the linktree.

### CODE REVIEW

File: content/en/smart-contracts/fundamentals/aggregators [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+This article explains the programmatic aggregated dealmaking process, via the aggregator standard. We will refer to small-scale data as 'subpiece data'.
+
+Aggregation can be done off-chain or on-chain. For on-chain aggregation smart contracts, developers interact with them to submit subpiece data and to verify proof of storage for their data. The contract also allows for verification of provided PoDSI, which refers to the subpiece proof. For off-chain aggregation, developers interact with the Aggregator platform instead, although all required components are managed by the platform for a simpler UI to clients. Typically, off-chain aggregation requires a centralized platform to host various components. On-chain aggregator can typically function in a decentralized manner.
+
+The base interface requires 4 components:
+- Client who has data to upload
+- Aggregator platform (a type of storage solution) that clients can interact with to make a deal and retrieve proof of storage
+- Off-chain aggregation node to aggregate the subpiece data into a larger file
+- On-chain Aggregation smart contract that clients can interact with to request to make a deal and verify proof of storage
+
```
Comment (trruckerfling) [inline content/en/smart-contracts/fundamentals/aggregators:30]: Adding this workflow chart for generic concept of aggregation

shapes (3)

File: content/en/smart-contracts/fundamentals/aggregators [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+3. Simultaneously, the aggregator aggregates indexed data segments (based on specs [here](https://github.com/filecoin-project/FIPs/discussions/512)). It runs the PoDSI proofing library and generates proofs for each subpiece pCID and stores it in an off-chain database. 
+
+4. The aggregator uses the [Boost API](https://boost.filecoin.io/experimental-features/fvm-contract-deals) (programmatic) or uses Lotus node (manual) to make storage deals with storage providers for the aggregated larger CAR file. 
+
+5. Storage Providers download the aggregated CAR file and publish storage deals. 
+
+6. Clients can query a proofing endpoint provided by the aggregator (example [here](https://docs.lighthouse.storage/lighthouse-1/filecoin-virtual-machine/podsi-and-deal-info), which will look up the subpiece CID (pCID) in the database and return the PoDSI proof, aggregated CID, and associated dealID.
+
+7. Clients can use the subpiece pCID for on-chain verification with the aggregation smart contract, which will verify merkle proof to ensure the pCID hash matches the commPa of the associated dealID. 
+
```
Comment (trruckerfling) [inline content/en/smart-contracts/fundamentals/aggregators:60]: Adding workflow diagram for off-chain aggregation

shapes 7

File: content/en/smart-contracts/fundamentals/aggregators [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+4. The aggregator combines the subpiece CIDs into the aggregated 32GiB CommP (CommPa) by computing within the aggregation smart contract.
+
+5. The aggregator uses the [Boost API](https://boost.filecoin.io/experimental-features/fvm-contract-deals) to make storage deals with storage providers for the aggregated larger CAR file. 
+
+6. Storage Providers download the aggregated CAR file and publish storage deals. Upon client's request, they can find the data via subpiece CID.
+
+6. Clients can query the aggregation smart contract, which notifies the aggregator to look up the subpiece CID (pCID) in its aggregation node's database and return the PoDSI proof, aggregated CID, and associated dealID.
+
+7. Simultaenously, clients can use the subpiece pCID for on-chain verification with the aggregation smart contract, which will verify merkle proof to ensure the pCID hash matches the commPa of the associated dealID. 
+
```
Comment (trruckerfling) [inline content/en/smart-contracts/fundamentals/aggregators:78]: Adding workflow diagram for on-chain aggregation

shapes 8

File: content/en/smart-contracts/fundamentals/aggregators [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+For uploading subpiece data for off-chain aggregation with a SDK, check out one of the implementations at [Lighthouse.storage](https://docs.lighthouse.storage/lighthouse-1/lighthouse-sdk/functions/upload).
+
+To build your own on-chain aggregator, check out one of the implementations with [Filecoin Data Tools](https://github.com/application-research/fevm-data-segment).
+
+### Aggregator + RaaS (Replication, Renew and Repair as a Service)
+
+Finally, both onchain and offchain aggregators can extend the use case beyond aggregation and dealmaking, to also cover renewal and replication services for stored data. Read more in this article.
```
Comment (trruckerfling) [inline content/en/smart-contracts/fundamentals/aggregators:85]: Need to hyperlink "article" to incoming RaaS doc

**Review by johnnymatthews — APPROVED** [review]

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+type: docs
+menu:
+  smart-contracts:
+    parent: "smart-contracts-fundamentals"
+    identifier: "aggregators"
+weight: 192
+toc: true
+---
+
+Aggregation is the process of combining multiple packages of _sub-piece data_ into a single package. This large package is stored on the Filecoin network instead of multiple smaller packages. Aggregation can be done off-chain or on-chain.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:16]: > Aggregation can be done off-chain or on-chain.

I think this is underspecified and can lead to confusion. The earlier sentence says that aggregation is "the process of combining [...]". That combination process won't happen on-chain, ever, as the chain never sees the data. It's possible to perform the combination on a metadata level (batching multiple subpieces in a single deal), but the way this opens is confusing to me.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+## Process
+
+The base interface for aggregation requires the following components:
+
+1. A client who has data to upload.
+1. An aggregator platform that clients can interact with to make a deal and retrieve proof of storage.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:23]: The usage of the term "deal" is confusing here. Assuming that the data referred to by (1) is a subpiece, the client does not make a deal with the aggregator. The client uploads the subpiece to the aggregator who, at its own volition, will eventually pack it into an aggregated deal.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+1. **Off-chain**: an aggregation node to aggregate the sub-piece data into a larger file
+
+    **On-chain**: an aggregation smart contract that clients can interact with to request to make a deal and verify proof of storage.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:26]: IMO this is not clear.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md (hunk 5 above)
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:16]: We should probably define what a subpiece is, explain why the Filecoin network cannot store subpieces as-is, and give examples of common subpieces (a video, NFT art, small dataset, etc.)

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+![](aggregation-generic.png)
+
+## Proof of Deal Sub-piece Inclusion (PoDSI)
+
+Proof of Deal Sub-piece Inclusion (PoDSI) is motivated by a need for sub-piece data uploads to contain verification and proof that the data was included in an associated deal on Filecoin. PoDSI is heavily used in the aggregated deal-making workflow.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:32]:
```suggestion
Proof of Deal Sub-piece Inclusion (PoDSI) is motivated by a need for sub-piece data uploads to eventually issue verification and proof that the data was included in an associated deal on Filecoin. PoDSI is heavily used in the aggregated deal-making workflow.
```

(The proof is not _contained_ in the upload)

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+PoDSI is a proofing library that enables the generation of a proof for each sub-piece CID (within the large data segment) and stores it in an off-chain database.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:34]: Incorrect. PoDSI is a proof construction, not a library. There are libraries that facilitate genreating a PoDSI, but PoDSI is not a library.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+The proof consists of two inclusion proofs:
+
+- An inclusion proof of a sub-tree corresponding to the tree of the client’s data. Optionally, this sub-tree can be misaligned with the proving tree.
+- An inclusion proof of the double leaf data segment descriptor within the data segment index.
+
+A sample [proof is available at lighthouse.storage](https://docs.lighthouse.storage/lighthouse-1/filecoin-virtual-machine/podsi-a-simple-overview).
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:41]: This does NOT link to a sample proof, but rather to instructions?

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+![](aggregation-generic.png)
+
+## Proof of Deal Sub-piece Inclusion (PoDSI)
+
+Proof of Deal Sub-piece Inclusion (PoDSI) is motivated by a need for sub-piece data uploads to contain verification and proof that the data was included in an associated deal on Filecoin. PoDSI is heavily used in the aggregated deal-making workflow.
+
+PoDSI is a proofing library that enables the generation of a proof for each sub-piece CID (within the large data segment) and stores it in an off-chain database.
+
+The proof consists of two inclusion proofs:
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:36]:
```suggestion
The proof consists of two elements:
```

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+
+- An inclusion proof of a sub-tree corresponding to the tree of the client’s data. Optionally, this sub-tree can be misaligned with the proving tree.
+- An inclusion proof of the double leaf data segment descriptor within the data segment index.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:39]:
```suggestion
1. An inclusion proof of a sub-tree corresponding to the tree of the client’s data. Optionally, this sub-tree can be misaligned with the proving tree.
1. An inclusion proof of the double leaf data segment descriptor within the data segment index.
```

Comment (raulk) [reply content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:39]: This goes into a lot of technical detail very quickly without even providing a link to expand information. Things like "data segment index", "double leaf", "data segment", "subtree", "misalignment" have not been explained prior to jumping into this.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+A sample [proof is available at lighthouse.storage](https://docs.lighthouse.storage/lighthouse-1/filecoin-virtual-machine/podsi-a-simple-overview).
+
+For further technical discussion, see the [Filecoin Improvement Proposal 512](https://github.com/filecoin-project/FIPs/discussions/512).
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:43]: Should link to the FRC too, which acts like the formal specification.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+
+### On-chain
+
+For on-chain aggregation smart contracts, developers interact with them to submit sub-piece data and verify proof of storage for their data. The contract also allows for verification of the provided PoDSI, which refers to the sub-piece proof.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:47]: Developers don't submit the subpiece data to the smart contract.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+
+Here is the process for on-chain aggregation:
+
+1. The client submits a sub-piece CID (CommPc) with metadata (e.g., metadata of the sub-piece and URL to download the CAR file) directly to the aggregation smart contract.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:51]: Can you open this explaining what the flow is trying to achieve?

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+1. The aggregator watches the aggregation contract, and when there are enough pieces to produce a 32GiB aggregated piece CID (CommPa), it downloads all sub-pieces for the aggregated piece from the CAR file URL.
+1. The aggregator aggregates indexed data segments into a larger data file for deal-making (based on specs [here](https://github.com/filecoin-project/FIPs/discussions/512)).
+1. The aggregator combines the sub-piece CIDs into the aggregated 32GiB CommP (CommPa) by computing within the aggregation smart contract.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:54]: The commP computation does not happen on chain.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+1. The aggregator uses the [Boost API](https://boost.filecoin.io/experimental-features/fvm-contract-deals) to make storage deals with storage providers for the aggregated larger CAR file.
+1. Storage Providers download the aggregated CAR file and publish storage deals. Upon the client's request, they can find the data via sub-piece CID.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:56]: Say that for the purposes of this flow, we won't elaborate on how this happens.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+1. Clients can query the aggregation smart contract, which notifies the aggregator to look up the sub-piece CID (pCID) in its aggregation node's database and return the PoDSI proof, aggregated CID, and associated deal ID.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:57]: Please expand on this. There are some technical inaccuracies here.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md (hunk 16 above)
Comment (raulk) [reply content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:54]: This is incorrect:
- Aggregated deal does not have to be 32GiB.
- Subpiece CIDs are not combined into a commP (it's the data).

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+1. Simultaneously, clients can use the sub-piece pCID for on-chain verification with the aggregation smart contract, which will verify the Merkle proof to ensure the pCID hash matches the commPa of the associated deal ID.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:58]: Wouldn't this be using the PoDSI, and not the pCID?

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+To build your own on-chain aggregator, check out one of the implementations with [Filecoin Data Tools](https://github.com/application-research/fevm-data-segment).
+
+### Off-chain
+
+For off-chain aggregation, developers interact with the aggregator platform instead, although all required components are managed by the platform for a simpler UI for clients. Typically, off-chain aggregation requires a centralized platform to host various components.
+
+Here is the process for off-chain aggregation:
+
+1. The client submits sub-piece data to an off-chain API. The API generates the sub-piece metadata and informs an Aggregator (e.g., sub-piece CID, known as pCID, and URL to download the CAR file).
+1. The aggregator hosts an off-chain aggregation node, which downloads these sub-piece CAR files and aggregates them into a larger CAR file.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:73]: Hm, this does not happen this way. Usually the user uploads a file which the aggregator formats into an IPLD DAG, usually turning it into a CAR for packing in a deal?

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 21]
```diff
@@ -0,0 +1,80 @@
...
+1. Simultaneously, the aggregator aggregates indexed data segments (based on specs [here](https://github.com/filecoin-project/FIPs/discussions/512)). It runs the PoDSI proofing library and generates proofs for each sub-piece pCID, storing them in an off-chain database.
+1. The aggregator uses the [Boost API](https://boost.filecoin.io/experimental-features/fvm-contract-deals) (programmatic) or uses Lotus node (manual) to make storage deals with storage providers for the aggregated larger CAR file.
+1. Storage Providers download the aggregated CAR file and publish storage deals.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:76]: This is oversimplified. There's many ways in which the transfer can happen.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md [hunk 22]
```diff
@@ -0,0 +1,80 @@
...
+1. Clients can query a proofing endpoint provided by the aggregator (example [here](https://docs.lighthouse.storage/lighthouse-1/filecoin-virtual-machine/podsi-and-deal-info), which will look up the sub-piece CID (pCID) in the database and return the PoDSI proof, aggregated CID, and associated deal ID.
+1. Clients can use the sub-piece pCID for on-chain verification with the aggregation smart contract, which will verify the Merkle proof to ensure the pCID hash matches the commPa of the associated deal ID.
```
Comment (raulk) [inline content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:78]: Once again, they would use the PoDSI?

**Review by raulk — CHANGES_REQUESTED** [review]

**Comment by johnnymatthews** [comment]
Switching to draft to make sure we don't accidentally merge this.

File: content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md (hunk 5 above)
Comment (hammertoe) [reply content/en/smart-contracts/fundamentals/aggregated-deal-making/index.md:16]: I agree. Maybe we need to say something like: "Aggregation can be initiated either via a smart contract or via the an off-chain SDK, however the actual aggregation will always take place off-chain"

**Comment by johnnymatthews** [comment]
This PR has been replaced with https://github.com/filecoin-project/filecoin-docs/pull/2201. Closing.

**Pull request closed without merging**